From aa67658af927241faa2e0d8850a3bf326f03466e Mon Sep 17 00:00:00 2001
Message-Id: <aa67658af927241faa2e0d8850a3bf326f03466e.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Daisy Sun <daisy.sun@intel.com>
Date: Mon, 21 Apr 2014 18:12:09 -0700
Subject: [PATCH 067/312] FOR_UPSTREAM [VPG]: drm/i915: Broaden FBC resolution
 limit to 4096*4096

Staring from BDW, the resolution limit of FBC has increased to
4096*4096

Issue: APDEV-2935
Otc-Tracker: VIZ-3826
Signed-off-by: Daisy Sun <daisy.sun@intel.com>
Reviewed-on: https://android.intel.com/187408
Reviewed-by: Azad, Vinit <vinit.azad@intel.com>
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>

Change-Id: I842f64e3cf2c0d18d29ef1bcfef3b9bb1f1764ac
Pulled from ABT/kernel-internal to GMin/gmin.

Signed-off-by: Oscar Mateo <oscar.mateo@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index bd66f88..df12130 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -534,7 +534,10 @@ void intel_update_fbc(struct drm_device *dev)
 		goto out_disable;
 	}
 
-	if (IS_G4X(dev) || INTEL_INFO(dev)->gen >= 5) {
+	if (INTEL_INFO(dev)->gen >= 8) {
+		max_width = 4096;
+		max_height = 4096;
+	} else if (IS_G4X(dev) || INTEL_INFO(dev)->gen >= 5) {
 		max_width = 4096;
 		max_height = 2048;
 	} else {
-- 
1.7.9.5

